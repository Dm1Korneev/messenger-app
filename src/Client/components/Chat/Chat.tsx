import CreateIcon from '@mui/icons-material/Create';
import IconButton from '@mui/material/IconButton';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';

import { ChatDto } from 'Types';

import { UsersAvatar } from '../UsersAvatar';

type Props = {
  selected: boolean;
  chatOnClick: () => void;
  chatModifyOnClick: (id: string) => void;
  chat: ChatDto;
}

export const Chat = ({
  selected, chatOnClick, chatModifyOnClick, chat,
}: Props) => {
  const { title, avatar, _id } = chat;

  return (
    <ListItemButton
      onClick={chatOnClick}
      selected={selected}
      sx={{
        '&:hover': {
          '& button': {
            visibility: 'visible',
          },
        },
      }}
    >
      <ListItemAvatar>
        <UsersAvatar author={title} avatar={avatar} />
      </ListItemAvatar>
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        size="large"
        sx={{ visibility: 'hidden', color: 'white' }}
      >
        <CreateIcon />
      </IconButton>
    </ListItemButton>
  );
};
