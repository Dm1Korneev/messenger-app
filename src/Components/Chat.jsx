import React from 'react';
import PropTypes from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import IconButton from '@material-ui/core/IconButton';

import CreateIcon from '@material-ui/icons/Create';

import UsersAvatar from 'Components/UsersAvatar';

function Chat(props) {
  const {
    selected, chatOnClick, chatModifyOnClick, classes, chat,
  } = props;
  const { title, avatar, _id } = chat;

  return (
    <ListItem
      button
      selected={selected}
      onClick={() => {
        chatOnClick(_id);
      }}
      className={classes.listItem}
    >
      <UsersAvatar author={title} avatar={avatar} />
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        classes={{
          root: classes.modifyButton,
        }}
      >
        <CreateIcon />
      </IconButton>
    </ListItem>
  );
}

Chat.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  selected: PropTypes.bool.isRequired,
  chatOnClick: PropTypes.func.isRequired,
  chatModifyOnClick: PropTypes.func.isRequired,
  chat: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
  }).isRequired,
};

const styles = () => ({
  listItem: {
    '&:hover': {
      '& button': {
        visibility: 'visible',
      },
    },
  },
  modifyButton: {
    visibility: 'hidden',
    color: '#FFF',
  },
});

export default withStyles(styles)(Chat);
