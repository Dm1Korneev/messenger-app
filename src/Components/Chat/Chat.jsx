import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import CreateIcon from '@material-ui/icons/Create';

import UsersAvatar from 'Components/UsersAvatar';

const useStyles = makeStyles(() => ({
  listItem: {
    '&:hover': {
      '& button': {
        visibility: 'visible',
      },
    },
  },
  modifyButton: {
    visibility: 'hidden',
    color: '#FFF',
  },
}));

const Chat = ({
  selected, chatOnClick, chatModifyOnClick, chat,
}) => {
  const classes = useStyles();

  const { title, avatar, _id } = chat;

  return (
    <ListItem
      button
      selected={selected}
      onClick={chatOnClick}
      className={classes.listItem}
    >
      <ListItemAvatar>
        <UsersAvatar author={title} avatar={avatar} />
      </ListItemAvatar>
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        classes={{
          root: classes.modifyButton,
        }}
      >
        <CreateIcon />
      </IconButton>
    </ListItem>
  );
};

Chat.propTypes = {
  selected: PropTypes.bool.isRequired,
  chatOnClick: PropTypes.func.isRequired,
  chatModifyOnClick: PropTypes.func.isRequired,
  chat: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
  }).isRequired,
};

export default Chat;

