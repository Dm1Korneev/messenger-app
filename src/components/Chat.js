import React from "react";
import PropTypes from "prop-types";

// components
import UsersAvatar from "./UsersAvatar";

// @material-ui
import withStyles from "@material-ui/core/styles/withStyles";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import CreateIcon from "@material-ui/icons/Create";
import IconButton from "@material-ui/core/IconButton";

function Chat(props) {
  const { selected, chatOnClick, chatModifyOnClick, classes, chat } = props;
  const { title, avatar, _id } = chat;

  return (
    <ListItem
      button
      selected={selected}
      onClick={() => {
        chatOnClick(_id);
      }}
      className={classes.listItem}
    >
      <UsersAvatar author={title} avatar={avatar} />
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        classes={{
          root: classes.modifyButton
        }}
      >
        <CreateIcon />
      </IconButton>
    </ListItem>
  );
}

Chat.propTypes = {
  classes: PropTypes.object.isRequired,
  selected: PropTypes.bool.isRequired,
  chatOnClick: PropTypes.func.isRequired,
  chatModifyOnClick: PropTypes.func.isRequired,
  chat: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired
  })
};

const styles = theme => ({
  listItem: {
    "&:hover": {
      "& button": {
        visibility: "visible"
      }
    }
  },
  modifyButton: {
    visibility: "hidden",
    color: "#FFF"
  }
});

export default withStyles(styles)(Chat);
