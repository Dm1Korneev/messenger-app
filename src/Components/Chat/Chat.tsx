import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import CreateIcon from '@material-ui/icons/Create';

import { Chat as ChatType } from 'Types';
import { UsersAvatar } from 'Components/UsersAvatar';

const useStyles = makeStyles(() => ({
  listItem: {
    '&:hover': {
      '& button': {
        visibility: 'visible',
      },
    },
  },
  modifyButton: {
    visibility: 'hidden',
    color: '#FFF',
  },
}));

type Props = {
  selected: boolean;
  chatOnClick: () => void;
  chatModifyOnClick: (id: string) => void;
  chat: ChatType;
}

export const Chat = ({
  selected, chatOnClick, chatModifyOnClick, chat,
}: Props) => {
  const classes = useStyles();

  const { title, avatar, _id } = chat;

  return (
    <ListItem
      button
      selected={selected}
      onClick={chatOnClick}
      className={classes.listItem}
    >
      <ListItemAvatar>
        <UsersAvatar author={title} avatar={avatar} />
      </ListItemAvatar>
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        classes={{
          root: classes.modifyButton,
        }}
      >
        <CreateIcon />
      </IconButton>
    </ListItem>
  );
};
