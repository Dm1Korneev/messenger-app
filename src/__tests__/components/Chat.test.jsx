import React from 'react';

import Chat from 'Components/Chat';

jest.mock('Components/UsersAvatar', () => global.mockComponent('UsersAvatar'));

jest.mock('@material-ui/core/ListItem', () => global.mockComponent('ListItem'));
jest.mock('@material-ui/core/ListItemText', () => global.mockComponent('ListItemText'));
jest.mock('@material-ui/core/IconButton', () => global.mockComponent('IconButton'));
jest.mock('@material-ui/icons/Create', () => global.mockComponent('Create'));

const props = {
  selected: false,
  chatOnClick: jest.fn(),
  chatModifyOnClick: jest.fn(),
  chat: {
    _id: '_id',
    title: 'title',
    avatar: 'avatar',
  },
};

describe('render Chat component', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = global.mount(
      <Chat {...props} />,
    );
  });

  test('component is render', () => {
    expect(wrapper.find('Chat').length).toBe(1);
  });

  test('ListItem subcomponent is render', () => {
    expect(wrapper.find({ originalcomponent: 'ListItem' }).length).toBe(1);
  });

  test('UsersAvatar subcomponent is render', () => {
    expect(wrapper.find({ originalcomponent: 'UsersAvatar' }).length).toBe(1);
  });
});

describe('snapshot-test Chat component', () => {
  test('Renders correct properties', () => {
    global.mountExpect(
      <Chat {...props} />,
    ).toMatchSnapshot();
  });
});
