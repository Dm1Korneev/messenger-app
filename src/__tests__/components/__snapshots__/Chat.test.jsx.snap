// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test Chat component Renders correct properties 1`] = `
<WithStyles(Chat)
  chat={
    Object {
      "_id": "_id",
      "avatar": "avatar",
      "title": "title",
    }
  }
  chatModifyOnClick={[MockFunction]}
  chatOnClick={[MockFunction]}
  selected={false}
>
  <Chat
    chat={
      Object {
        "_id": "_id",
        "avatar": "avatar",
        "title": "title",
      }
    }
    chatModifyOnClick={[MockFunction]}
    chatOnClick={[MockFunction]}
    classes={
      Object {
        "listItem": "Chat-listItem-1",
        "modifyButton": "Chat-modifyButton-2",
      }
    }
    selected={false}
  >
    <Component
      button={true}
      className="Chat-listItem-1"
      onClick={[Function]}
      selected={false}
    >
      <div
        originalcomponent="ListItem"
        testprops={
          Object {
            "button": true,
            "children": Array [
              <Unknown
                author="title"
                avatar="avatar"
              />,
              <Unknown
                primary="title"
              />,
              <Unknown
                classes={
                  Object {
                    "root": "Chat-modifyButton-2",
                  }
                }
                onClick={[Function]}
              >
                <Unknown />
              </Unknown>,
            ],
            "className": "Chat-listItem-1",
            "onClick": [Function],
            "selected": false,
          }
        }
      >
        <Component
          author="title"
          avatar="avatar"
        >
          <div
            originalcomponent="UsersAvatar"
            testprops={
              Object {
                "author": "title",
                "avatar": "avatar",
              }
            }
          />
        </Component>
        <Component
          primary="title"
        >
          <div
            originalcomponent="ListItemText"
            testprops={
              Object {
                "primary": "title",
              }
            }
          />
        </Component>
        <Component
          classes={
            Object {
              "root": "Chat-modifyButton-2",
            }
          }
          onClick={[Function]}
        >
          <div
            originalcomponent="IconButton"
            testprops={
              Object {
                "children": <Unknown />,
                "classes": Object {
                  "root": "Chat-modifyButton-2",
                },
                "onClick": [Function],
              }
            }
          >
            <Component>
              <div
                originalcomponent="Create"
                testprops={Object {}}
              />
            </Component>
          </div>
        </Component>
      </div>
    </Component>
  </Chat>
</WithStyles(Chat)>
`;
